internals/enterprise_helpers.py

```diff
@@ -0,0 +1,148 @@
+from internals.core_models import FeatureEntry
+
+CHANNEL_DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%S'
+
+def _str_to_datetime(dt):
+  return datetime.strptime(dt, CHANNEL_DATETIME_FORMAT)
+
+def needs_default_first_notification_milestone(
+    existing_feature: FeatureEntry | None = None,
+    new_fields: dict = {}) -> bool:
+  """Returns whether we should create a default value for first_enterprise_notification_milestone.
+
+  If a feature required a first_enterprise_notification_milestone field and a valid value is not
+  available, we generate a deafult one.
+
+  Args:
+    existing_feature: FeatureEntry Optional feature that needs to be updated.
+    new_fields: dict Fields that will be used to update or create the feature.
+  """
+  milestone = None
+  has_valid_milestone_in_new_fields = False
+  if ('first_enterprise_notification_milestone' in new_fields and
+      new_fields['first_enterprise_notification_milestone']):
+    milestone = int(new_fields['first_enterprise_notification_milestone'])
+    channel_details = channels_api.construct_specified_milestones_details(milestone, milestone)
+    has_valid_milestone_in_new_fields = (
+      milestone in channel_details and
+      _str_to_datetime(channel_details[milestone]['stable_date']) > datetime.now())
+
+  # We are creating a new feature
+  if existing_feature is None:
+    # All enterprise features need this
+    if new_fields['feature_type'] == FEATURE_TYPE_ENTERPRISE_ID:
+      return not has_valid_milestone_in_new_fields
+    # All breaking changes need this
+    if 'breaking_change' in new_fields and new_fields['breaking_change'] == True:
+      return not has_valid_milestone_in_new_fields
+    return False
+
+  # We are updating a feature that already has a notification milestone
+  if existing_feature.first_enterprise_notification_milestone != None:
+    return False
+  
+  # The enterprise feature we are updating does not have the field
+  if existing_feature.feature_type == FEATURE_TYPE_ENTERPRISE_ID:
+    return not has_valid_milestone_in_new_fields
+  
+  # The breaking change stays a breaking change
+  if existing_feature.breaking_change == True and ('breaking_change' not in new_fields or
+                                                    new_fields['breaking_change'] == True):
+    return not has_valid_milestone_in_new_fields
+  
+  # The normal feature becomes a breaking change
+  if 'breaking_change' in new_fields and new_fields['breaking_change'] == True:
+    return not has_valid_milestone_in_new_fields
+  
+  return False
+
+def can_update_first_notification_milestone(feature: FeatureEntry, new_fields: dict) -> bool:
+  """Returns whether the milestone can be used to update first_enterprise_notification_milestone.
+
+  Args:
+    existing_feature: FeatureEntry feature that needs to be updated.
+    new_fields: dict Fields that will be used to update or create the feature.
+  """
+  if ('first_enterprise_notification_milestone' not in new_fields or
```

Comment:
These few lines of code could be simpler as:
milestone = new_fields.get('first_enterprise_notification_milestone')
if not milestone:
  return False